src/Config/IslandoraSolrMetadataFieldConfig.php

```diff
@@ -0,0 +1,224 @@
+<?php
+
+namespace Drupal\islandora_solr_metadata\Config;
+
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Component\Utility\SortArray;
+
+/**
+ * Metadata display profile field config.
+ *
+ * Provides implementing code a method of accessing field info by using
+ * actual Solr field names.
+ */
+class IslandoraSolrMetadataFieldConfig {
```

Comment:
Really, being defined as a service... we should probably have the public API here defined by an interface, which we then implement...

... which is to say, define an interface, and then use it in all places except defining the default service, instead of referring to this class explicitly.